lib/puppet/indirector/catalog/rest.rb

```diff
@@ -13,6 +13,14 @@ def find(request)
 
     session = Puppet.lookup(:http_session)
     api = session.route_to(:puppet)
+
+    ip_address = begin
+      " (#{Resolv.new.getaddress(api.url.host)})"
+    rescue Resolv::ResolvError
```

Comment:
I think we only need to call the `getaddress` class method here, so we can just do `Resolve.getaddress`.

I'm not sure if we need to rescue Resolv::ResolvError, is there a case where `route_to` would validate a server but that server would fail to resolve to an IP address later?

Reply:
`route_to` may return a cached service object (that it created earlier in the agent lifecycle to do pluginsync, etc) https://github.com/puppetlabs/puppet/blob/6f19bb8a6d6d48b10bf92cb3476804175b6992e1/lib/puppet/http/session.rb#L60-L61 so it's unlikely but possible the call to `getaddress` could fail.